In [108]:
import tensorflow as tf
import numpy as np
m = tf.keras.metrics.Accuracy()
y_true = np.array([
    [0,1,0,0], # 1st row
    [1,0,0,0], # nth row
    [0,1,0,0],
])
y_pred = np.array([
    [0,1,0,0],
    [1,0,0,0],
    [1,0,0,0],
])
m.update_state(y_true, 
               y_pred)
m.result().numpy()

0.8333334

In [121]:
m.reset_state()
m.update_state(tf.argmax(y_true, axis=-1), 
               tf.argmax(y_pred, axis=-1))
m.result().numpy()

0.6666667

There are total 12 lemeents out of which 10 are correct `10 / 12 = 0.8333`

In [125]:
import tensorflow as tf
import numpy as np
m = tf.keras.metrics.CategoricalAccuracy()
y_true = np.array([
    [0,1,0,0], # 1st row
    [1,0,0,0], # nth row
    [0,1,0,0],
])
y_pred = np.array([
    [0,1,0,0],
    [1,0,0,0],
    [1,0,0,0],
])
m.update_state(y_true, 
               y_pred)
m.result().numpy()

0.6666667

In [105]:
import tensorflow_addons as tfa
metric = tfa.metrics.F1Score(
    num_classes=4, threshold=0.5)
y_true = np.array([
    [0, 1, 1, 0],
    [0, 0, 0, 1],
    [1, 0, 1, 0],
])
y_pred = np.array([
    [0, 1, 0, 0],
    [0, 1, 0, 0],
    [0.6, 0, 0.51, 0],
])
metric.update_state(y_true, y_pred)
result = metric.result()
result.numpy()

array([1.       , 0.6666667, 0.6666667, 0.       ], dtype=float32)

```
       class 0  ,   class 1,   class 2, class 3
array([1.       , 0.6666667, 0.6666667, 0.       ], dtype=float32)
```

3. class 2
   1. precision = 1 / 1 + 0
   2. recall    = 1 / 1 + 1
   3. f1 score  = 2 * 1 * 0.5 / (1 + 0.5) = 0.666
4. class 3
   1. precision = 0 / 0 + 0
   2. recall    = 0 / 0 + 1
   3. f1 score  = 2 * 0 * 0 / (0 + 0) = 0

In [75]:
from sklearn.metrics import f1_score
y_true = [0, 1, 2, 0, 1, 2]
y_pred = [0, 2, 1, 0, 0, 1]
f1_score(y_true, y_pred, average=None)

array([0.8, 0. , 0. ])